lib/components/util/formatted-day-of-week-compact.js

```diff
@@ -0,0 +1,35 @@
+import { FormattedMessage } from 'react-intl'
+import PropTypes from 'prop-types'
+
+/**
+ * Returns a FormattedMessage component for the compact day of week such that i18n IDs
+ * are hardcoded and can be kept track of by format.js tools
+ */
+export default function FormattedDayOfWeekCompact ({day}) {
+  switch (day) {
```

Comment:
If template string id generation is considered acceptable, this would be a great opportunity to make things more compact. 

It looks like PropTypes can be used to make sure that the input is an "allowed" string, even if we don't have typescript in this file yet: https://stackoverflow.com/a/60912152. This should resolve all qualms about using template strings for id generation